sdk/python/kfp/components/importer_node.py

```diff
@@ -54,6 +54,14 @@ def pipeline_with_importer():
               reimport=False)
           train(dataset=importer1.output)
     """
+    if issubclass(artifact_class, artifact_types.Artifact
+                 ) and not artifact_class.TYPE_NAME.startswith('system.'):
+        # For artifact classes not under the `system` namespace,
+        # use its TYPE_NAME as-is.
+        type_name = artifact_class.TYPE_NAME
+    else:
+        type_name = artifact_class.__name__
```

Comment:
I'm curious what the value of using/permitting `artifact_class.__name__` is. It seems like we'd want to allow (and even enforce) artifact class authors to set the schema_title using an attribute rather than the class name, especially for non-system artifacts.

This is thinking ahead a little bit, but I think we want to soon move away from checking if a class is an artifact class by checking if it is a subclass of `dsl.Artifact` and instead just checking that it adheres to the KFP artifact class protocol (which includes having a `.TYPE_NAME` attribute), that way third-party libraries containing KFP artifacts do not need to depend on KFP. This would require that artifact classes have a `TYPE_NAME` attribute.

cc: @chensun

Reply:
Following up on my own question after more investigation.

Use of `artifact_class.__name__` effectively strips the `'system.'` for KFP-native artifacts (since the class name should be equal to `.TYPE_NAME` after `'system.'` is stripped. This is likely done this way for compatibility with KFP v1: an artifact is encoded in v1 component YAML as `'Artifact'`, not `'system.Artifact'`.

My preference is to preserve the full type name (with `'system.'`) to reduce any ambiguity in naming and enable backward-compatibility with v1 through logic in v2, not through the compressed artifact name representation.